src/Gren/Kernel/HttpServer.js

```diff
@@ -0,0 +1,106 @@
+/*
+
+import Gren.Kernel.Scheduler exposing (binding, succeed, fail, rawSpawn)
+import HttpServer exposing (ServerError, toRequest)
+import HttpServer.Response as Response exposing (toResponse)
+import Platform exposing (sendToApp, sendToSelf)
+
+*/
+
+// TODO: https support?
+const http = require("http");
+
+// TODO: Haven't thought about concurrent servers/requests yet.
+// At least need to add unique identifiers.
+
+var _HttpServer_createServer = function (options) {
+  return __Scheduler_binding(function (callback) {
+    const server = http.createServer();
+    server.on("error", function (e) {
+      callback(
+        __Scheduler_fail(A2(__HttpServer_ServerError, e.code, e.message))
+      );
+    });
+    server.listen(options.port_, options.host, function () {
```

Comment:
Thanks. It's also crashing with optimize for other reasons. After removing kernel code and incrementally re-adding it I think I've narrowed it down to [my call to `__HttpServer_toRequest`](https://github.com/blaix/gren-lang-node/pull/1/files#diff-b6d58b86d734deb492ee4940ea3eb6f1ebe84b5d1ef1e484019b1733ecc4e054R51-R61) but I haven't figured out why yet.

Reply:
Have you tried prefixing the properties of the object you're sending in with `__$`?

Gren code will be trying to access properties with `obj.a` instead of `obj.urlProtocol` in optimized mode.

Reply:
Oh wow, I completely missed that it needs to be both ways. So if an object is coming from gren I need to access properties with the prefix, and if it's going to gren I need to send the properties with the prefix. That did the trick once I fixed a couple other places as well. Thanks!